### ISSUE TITLE
1st-party rule allows subdomains by default

### ISSUE THREAD

**Issue by g4jc** [gorhill/uMatrix#278]
Hello,
This was my first time using uMatrix, and I must say it is an excellent work in progress. Thanks for your work!

However, one of my biggest issues thusfar, is the 1st-party rule. I had configured it to block all, but allow first party CSS and Images. Which worked for the most part.

But I couldn't find a way to block subdomains. As a test case:
https://fsf.org/ --> Images come from static.fsf.org, but are green. piwik.fsf.org logs, etc.
There are also sites that use subdomains for their users e.g. goodblog.somefreehost.com could get hacked by badblogcommenter.somfreehost.com 
So I like to control all my configurations at the subdomain level when applying rules.

In Summary: Is it possible to apply a generic rule to only allow the current tab's root domain by default, but block any requests to any of it's subdomains?

**Comment by gorhill** [comment]
> Is it possible to apply a generic rule to only allow the current tab's root domain by default, but block any requests to any of it's subdomains?

Not possible. I don't plan to support this, this is not merely an extra feature, this means completely changing uMatrix core.
